### PR TITLE
cake5: refactor annotate shell to commands

### PR DESCRIPTION
**Pull request by LordSimal**
Closes https://github.com/dereuromark/cakephp-ide-helper/issues/311

Since Subcommands aren't a thing any more, I extracted each functionality into its own command and manually mapped it in the `Plugin::console()` method

### CODE REVIEW

**Comment by LordSimal** [comment]
well, it seems like I have to do the PHPUnit 10 upgrade here as well...

File: src/Command/Annotate/AllCommand.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+				$io->out('[' . $shortName . ']');
+			}
+			if (!$this->interactive) {
+				$in = 'y';
+			} else {
+				$in = $io->askChoice($type . '?', ['y', 'n', 'a'], 'y');
+			}
+
+			if ($in === 'a') {
+				$this->abort('Aborted');
```
Comment (dereuromark) [inline src/Command/Annotate/AllCommand.php:70]: do we have to separate this into 2 lines?

File: src/Command/Annotate/CallbacksCommand.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+use IdeHelper\Annotator\CallbackAnnotator;
+use IdeHelper\Command\AnnotateCommand;
+
+class CallbacksCommand extends AnnotateCommand {
+
+	/**
+	 * @return string
+	 */
+	public static function getDescription(): string {
+		return 'Annotate callback methods using callback annotation tasks. This task is not part of "all" when "-r" is used.';
```
Comment (dereuromark) [inline src/Command/Annotate/CallbacksCommand.php:19]:
```suggestion
		return 'Annotate callback methods using callback annotation tasks. This task is not part of `all` when `-r` is used.';
```

**Comment by dereuromark** [comment]
Wow, great start here. A few changes happened in core since the last time I touched this as well.

**Comment by LordSimal** [comment]
The only thing left here is the missing plugins which I have no idea why this worked before

**Comment by LordSimal** [comment]
I seem to have found the reason: With the current `master` branch the plugin-installer gets installed but with `cake5` it doesn't.

**Pull request merged**